palm/plugins/dbt/commands/cmd_model-doc.py

```diff
@@ -0,0 +1,215 @@
+import click
+import sqlparse
+import re
+import yaml
+from pathlib import Path
+from functools import lru_cache
+from typing import List
+
+
+@click.command("model-doc")
```

Comment:
We have an existing command group `model` which has a command `new` for generating models from boilerplate/template files... Could we add this under that group? Like `palm model docs-generate`?

Reply:
Or just `palm model docs`

Reply:
Hyphenated command names feel messy to me to have available in our public package, as opposed to local dbt project commands we spin up on the fly which are just for us.

If we are unhappy with our prior implementation of generating model docs, maybe we should remove that and replace it with this?

Reply:
Having this as a separate command serves both the ability to create docs retrospectively (e.g., palm model-docs my_old_model) and can be added to the existing command group as an option (or otherwise).

Reply:
Naming things is hard, but short of `palm docgen` I can't think of a better name for this command.

Hyphenated commands are very common in CLIs, `dbt` even has `run-operation`. I agree, single-word command names are preferable, because they are shorter to type and easier to remember. However, I think we're already in a weird spot on this for a couple of reasons:
- Adding the command to the `model` group is actually more typing, and the functionality kind of gets hidden in the `model` command group.
- We have `palm dbt-docs` which generates dbt docs and runs it on localhost, this could easily be confused with the new command because they're both `LIKE %docs%`

Changing how `palm model new` works is a bit outside of the scope of the ticket. I'll take a look at the implementation tomorrow and see what that might look like, and if there's a simple way to handle both cases together. I know, right now, there isn't 1:1 functionality because the other implementation used comments in the ref file to build out docs, where my approach deals with existing doc files and then falls back to TODOs.

Question for the group: has anyone used `palm model new` with the ref file recently?

Reply:
I looked at the way `model new` generates docs, there's an opportunity to share some of that logic but it's not super simple and the refactor will be risky (untested code). I think for now we can keep this separate and iterate on it later.